process/blockProcessor.go

```diff
@@ -15,22 +16,34 @@ const (
 )
 
 type blockProcessor struct {
-	proc     Processor
-	dbReader ExternalStorageConnector
+	proc                                      Processor
+	dbReader                                  ExternalStorageConnector
+	getLatestFullySynchronizedHyperblockNonce func() (uint64, error)
 }
 
 // NewBlockProcessor will create a new block processor
-func NewBlockProcessor(dbReader ExternalStorageConnector, proc Processor) (*blockProcessor, error) {
+func NewBlockProcessor(
+	dbReader ExternalStorageConnector,
+	proc Processor,
+	getLatestFullySynchronizedHyperblockNonce func() (uint64, error),
+) (*blockProcessor, error) {
 	if check.IfNil(dbReader) {
 		return nil, ErrNilDatabaseConnector
 	}
 	if check.IfNil(proc) {
 		return nil, ErrNilCoreProcessor
 	}
+	// if function is nil will return always MaxUint64 -> result of function will be ignored
+	if getLatestFullySynchronizedHyperblockNonce == nil {
+		getLatestFullySynchronizedHyperblockNonce = func() (uint64, error) {
```

Comment:
I would rather return error.
If someone does not care about fully sinchronized blocks, then we should implement an option that passes this mock to to newBlockProcessor.

Reply:
done.